subsys/bluetooth/services/ancs_client.c

```diff
@@ -368,15 +364,13 @@ static void bt_ancs_cp_write_callback(struct bt_conn *conn, uint8_t err,
 
 	atomic_clear_bit(&ancs_c->state, ANCS_CP_WRITE_PENDING);
 
-	struct bt_ancs_evt ancs_evt;
-
-	ancs_evt.evt_type = BT_ANCS_EVT_NP_ERROR;
-	ancs_evt.err_code_np = err;
-
-	ancs_c->evt_handler(&ancs_evt);
+	if (ancs_c->cp_write_cb) {
+		ancs_c->cp_write_cb(ancs_c, err);
+	}
 }
 
-int bt_ancs_cp_write(struct bt_ancs_client *ancs_c, uint16_t len)
+int bt_ancs_cp_write(struct bt_ancs_client *ancs_c, uint16_t len,
+		     bt_ancs_write_cb func)
```

Comment:
Also the function is public, but could potentially conflict with ANCS_CP_WRITE_PENDING in progress also using the write parameters, right?

Reply:
bt_ancs_cp_write() is a routine to minimize repeated code.

There are several calls to this function. Each sets the ANCS_CP_WRITE_PENDING bit to claim usage of GATT write parameters and a common data buffer. When the GATT write is not done, ANCS_CP_WRITE_PENDING is set and further calls to set the bit gets a -EBUSY error. The function prototype gets a comment to document this.